build and golden masters for tests

NolaDonato opened the pull request:
1. Tests use common.gradle
2. Added delays to make render tests more reliable
3. Fixed path to golden masters

SXR DCO signed off by: Nola Donato nola.donato@samsung.com

NolaDonato committed:
Tests use common.gradle

Added delays to make render tests more reliable
Fixed path to golden masters

NolaDonato committed:
Update build for asset tests, used 1Kx1K display

adding missing golden masters

NolaDonato committed:
1. put VIdeoTests and WidgetTests in their own sections
2. remove VertexColorShader, PhongShader now supports vertex colors

thomasflynn commented:
cherry-picked this pull-request.  cannot build.  get:
* Where:
Build file '/home/flynnt/trees/sxr/sxrsdk-tests/asset-tests/app/build.gradle' line: 45

* What went wrong:
A problem occurred evaluating project ':asset-tests'.
> Could not find method implementationDebug() for arguments [{name=x3d-debug, ext=aar}] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

NolaDonato committed:
make framework tests more resilient to intermittent failures

1. Dont make picker tests require exact frame counts to pass, allow +- 1 frame to allow for timing differences
2. Log start/stop of picker tests to diagnose hangs
3. Work around timing issues in RenderConfigTests

NolaDonato committed:
Fix test build to allow building against non-test version

Test build.gradle assumed the project had "useLocalDependencies" set.
If that is not set, the tests will run against current version

NolaDonato commented:
I could not reproduce your problem but nevertheless tried to fix it by looking at other build.gradle files. My shot in the dark has been committed, please try to see if it builds for you now.

NolaDonato committed:
Move stress test assets to correct directory

NolaDonato committed:
Moved testSendEvents to io category

NolaDonato committed:
blank golden masters

NolaDonato committed:
Wait a frame after centering model in tests

NolaDonato committed:
add waits to allow tests more time before screenshot

NolaDonato committed:
add waits for asset tests, add missing golden masters

NolaDonato committed:
put back x3d tests

NolaDonato committed:
add waits in asset import tests

NolaDonato committed:
fix texture count in test

NolaDonato committed:
add delays in in external scene test

liaxim commented:
@thomasflynn Currently the tests build if the tests and the x3d and the script extensions are part of the framework workspace. Will look into fixing the standalone build.

liaxim commented:
Depends on https://github.com/sxrsdk/sxrsdk/pull/61

liaxim commented:
The tests are in decent shape. TestSendEvents crashes for me.

liaxim commented on framework-tests/app/src/androidTest/java/com/samsungxr/stress/StressTests.java:
```diff
@@ -140,18 +141,20 @@ public void gcOomTest2() throws Exception {
                 SXRNode sceneObject = new SXRNode(sxrContext, sxrContext.createQuad(10f, 10f));
                 SXRRenderData renderData = sceneObject.getRenderData();
-                SXRTexture texture;
+                texture.clearData();
```
I find this clearData confusing. Immediately a new instance will be stored in "texture". If all is good, the old texture will be destroyed and dtor calls clearData. If there is really a need to call clearData explicitly here, it is a sign that something is not working as intended.

liaxim commented:
I think the tests should not go into dedicated packages. They should share the main package as the framework to avoid exposing internal methods, like clearData, as public. There shouldn't be internal methods ideally but if we must, they better be hidden from the public API space.
